src/Stratis.Bitcoin.Features.MemoryPool/Rules/CheckTxOutDustRule.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.Policy;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+
+namespace Stratis.Bitcoin.Features.MemoryPool.Rules
+{
+    /// <summary>
+    /// This rule ensures that the transaction does not contain any outputs which are considered dust.
+    /// </summary>
+    public sealed class CheckTxOutDustRule : MempoolRule
+    {
+        public CheckTxOutDustRule(
+            Network network,
+            ITxMempool mempool,
+            MempoolSettings mempoolSettings,
+            ChainIndexer chainIndexer,
+            ILoggerFactory loggerFactory) : base(network, mempool, mempoolSettings, chainIndexer, loggerFactory)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void CheckTransaction(MempoolValidationContext context)
+        {
+            foreach (var txOut in context.Transaction.Outputs)
```

Comment:
Before the loop we should check if `vin.Count > vout.Count`; reducing total set of UTXO is a good thing for the network regardless of the amounts involved.
`if (vin.Count > vout.Count) break;`

Reply:
I don't see how this matter for dust.

Reply:
@dangershony the decision here is to reduce the amount of UTXOs if the inputs are more than the outputs. We are effectively consolidating dust utxo's on the network by allowing this transaction to pass.

Reply:
If you have 20 dust inputs and a dust output its still dust no?

Reply:
From the link above
```
Bitcoin Core considers a transaction output to be dust, when its value is lower than the cost of spending it at the dustRelayFee rate.
```

I think this should be the only indicator to disallow relay of dust and not number of outputs or inputs.

Reply:
Yes, you are reducing 20 dust UTXO to one dust UTXO. As a result there is less UTXOs which is a good thing

Reply:
That is true but then we decided that it is user's issue then. If he/she wants to submit a transaction like that we should allow it, even it costs them more to send it. At least we are gaining from a smaller UTXO set.

This is a highly unlikely scenario though ;)